### DISCUSSION TITLE
Fleet shows no hosts - Win10 osquery msi not connecting

### DISCUSSION THREAD

**Discussion by innovate-support** [Security-Onion-Solutions/securityonion#2317] [Unsupported Versions]
My SO 2.3.2 server seems to be working fine except for Fleet.  I don't see any hosts at all in Fleet, not even the SO Server. I've run so-allow multiple times to allow not only the specific IP of the Win10 machine, but also the whole network range. I can nmap from Win10 and see that port 8090 is open on the SO Server, but no hosts ever show up in Fleet even after reinstalling the msi-launcher.msi and rebooting both the Win10 machine and the SO Server. Both are VM's that are on the same virtual network and are talking fine with each other over ports 80, 443, and 22. All other tools seem to be working fine with Alerts, but no osquery logs are coming in.

I am running the server in Standalone mode.
sudo so-status shows all OK

**Comment by defensivedepth** [comment]
I would start off by upgrading to 2.3.10 via soup; we fixed a few osquery/Fleet issues with 2.3.10. 

If you still don't see the SO Server in Fleet after the upgrade, please run the following on the Manager and post the output:

`sudo salt-call state.apply fleet.install_package`

`sudo systemctl status launcher.so-launcher.service -l`

**Reply by innovate-support** [reply]
- Updated Server to v2.3.10 using soup
- checked so-status and all shows OK
- Redownloaded the launcher.msi and ran it on Win10 host
- ran nmap to confirm Win10 can communicate to Server on port 8090, all OK
- Rebooted Win10 for good measure
Still no hosts appearing in Fleet on Server

- ran the requested salt-call command and got this red output
**local:
    Data failed to compile:
----------
    The function "state.highstate" is running as PID 9271 and was started at 2020, Dec 11 22:03:08.556261 with jid 20201211220308556261**

- ran the systemctl command and it started osqueryd successfully
- ran the salt-call command again and now it shows the following green ouput
**local:
----------
          ID: launcherpkg
    Function: pkg.installed
      Result: True
     Comment: All specified packages are already installed
     Started: 22:07:50.564225
    Duration: 1104.868 ms
     Changes:   
Summary for local
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
Total run time:   1.105 s**

- Fleet still doesn't show any hosts
- tried running the msi on Win10 host again but nothing changes

What exactly is supposed to be happening/running on the Windows machine after installing the msi?  All I see is that there's a Kolide folder with the expected conf and data files added in C:\Program Files.  Is it supposed to add a Service or some Processes, or something in the Start menu?  And if so, what are they called?  Is there some exe or command I need to be running on the Windows machine to test osquery communication/functionality?

**Reply by innovate-support** [reply]
I can watch launcher.exe run LauncherSOLauncherSvc and then disappear every 45 sec or so in the Windows Process window.  Looks like it's doing something intentional.

**Reply by innovate-support** [reply]
So after digging around some more I seem to have stumbled upon a fix.  This seems like it may be a bug in Fleet.

My particular SO Server installation method uses FQDN as it's communication type.
My Win10 machine is on the same IP subnet as the SO Server, and it can also see and ping and communicate with the server using it's FQDN with no issues.  In the **C:\Program Files\Kolide\Launcher-so-launcher\conf\launcher.flags** file that was auto installed it shows the hostname of the Server correctly.  So, since I know sometimes these things can be finicky I went ahead and change the FQDN shown in that **launcher.flags** file to the Server's IP instead.  Within 30 seconds of clicking Save the Win10 host magically appeared in the Fleet web interface.

**Reply by innovate-support** [reply]
Please mark this as Answered.  Changing the FQDN shown in the **C:\Program Files\Kolide\Launcher-so-launcher\conf\launcher.flags** file to the Server's IP resolved the issue.

**Reply by defensivedepth** [reply]
@innovate-support Trying to dig a little deeper on this issue to ascertain a root cause.

Can you post the output of the following, sanitizing as needed? (run on the Manager)

`sudo salt-call pillar.get global`

**Reply by innovate-support** [reply]
local:
    ----------
    airgap:
        False
    cortexkey:
        xxxxx
    cortexorgname:
        SecurityOnion
    cortexorguserkey:
        xxxxx
    cortexplaysecret:
        xxxxx
    fleet_custom_hostname:
        None
    fleet_hostname:
        so
    fleet_ip:
        xxxxx
    fleet_manager:
        True
    fleet_node:
        False
    fleet_packages-timestamp:
        2020-12-22-15:49
    fleet_packages-version:
        3
    hivekey:
        xxxxx
    hiveplaysecret:
        xxxxx
    hnmanager:
        xxxxx
    ids:
        Suricata
    imagerepo:
        securityonion
    managerip:
        xxxxx
    managerupdate:
        1
    mdengine:
        ZEEK
    ntpserver:
    pipeline:
        redis
    proxy:
    sensoronikey:
        xxxxx
    soversion:
        2.3.21
    url_base:
        xxxxx
    wazuh:
        1

**Reply by defensivedepth** [reply]
@innovate-support Is `url_base` from `sudo salt-call pillar.get global` your hostname or IP?

**Reply by dylanbegin** [reply]
> 
> 
> So after digging around some more I seem to have stumbled upon a fix. This seems like it may be a bug in Fleet.
> 
> My particular SO Server installation method uses FQDN as it's communication type.
> My Win10 machine is on the same IP subnet as the SO Server, and it can also see and ping and communicate with the server using it's FQDN with no issues. In the **C:\Program Files\Kolide\Launcher-so-launcher\conf\launcher.flags** file that was auto installed it shows the hostname of the Server correctly. So, since I know sometimes these things can be finicky I went ahead and change the FQDN shown in that **launcher.flags** file to the Server's IP instead. Within 30 seconds of clicking Save the Win10 host magically appeared in the Fleet web interface.

I had this same exact issue and was able to fix it by doing this as well, thank you!!! My url_base is FQDN as that is what I remember setting during install.

Also, not sure if this is relevant but the fleet_hostname: section of the pillar global.sls file only had the hostname and not the FQDN.

**Reply by innovate-support** [reply]
my `url_base` shown by `sudo salt-call pillar.get global` is my Manager's FQDN

**Reply by em-sec** [reply]
Resolved the same issue on mine. I can confirm my url_base is FQDN and my fleet_manager is hostname only.

**Comment by defensivedepth** [comment]
If this is still an issue, please let me know.

**Reply by facyber** [reply]
Hi @defensivedepth 

I have this same issue. I am using the latest SO 2.3.40, I tried same steps provided, to set IP instead of FQDN and still don't see nodes in the fleet. I checked tcpdump on port 8090 on Manager node and I can confirm my agents communicate. I tried both Linux and Windows hosts, still same.

Result of `salt-call pillar.get global`:

```
local:
    ----------
    airgap:
        False
    cortexkey:
        xxxxxxxxxxxxx
    cortexorgname:
        SecurityOnion
    cortexorguserkey:
        xxxxxxxxxxxxx
    cortexplaysecret:
        xxxxxxxxxxxxx
    dockernet:
        xxxxxxxxxxxxx
    fleet_custom_hostname:
        None
    fleet_hostname:
        manasgernode01
    fleet_ip:
        manager_ip_address
    fleet_manager:
        True
    fleet_node:
        False
    fleet_packages-timestamp:
        2021-04-02-08:20
    fleet_packages-version:
        3
    hivekey:
        xxxxxxxxxxxxx
    hiveplaysecret:
        xxxxxxxxxxxxx
    hnmanager:
        10.0.0.0/8,192.168.0.0/16,172.16.0.0/12
    ids:
        Suricata
    imagerepo:
        security-onion-solutions
    managerip:
        manager_ip_address
    managerupdate:
        1
    mdengine:
        ZEEK
    ntpserver:
    pipeline:
        redis
    proxy:
    sensoronikey:
        xxxxxxxxxxxxx
    soversion:
        2.3.40
    url_base:
        security_onion_dns
    wazuh:
        1
```

I also executed successfully `salt-call state.apply fleet.install_package` and here is the result from `systemctl status launcher.so-launcher.service -l`:

```
Apr 05 09:32:30 MANAGER_HOSTNAME launcher[12846]: {"caller":"updater.go:112","err":"enrolling host: transport error in enrollment: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: authentication handshake failed: x509: certificate signed by unknown authority\"","msg":"updater interrupted","severity":"info","ts":"2021-04-05T09:32:30.153273843Z"}
Apr 05 09:32:30 MANAGER_HOSTNAME launcher[12846]: {"caller":"updater.go:112","err":"enrolling host: transport error in enrollment: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: authentication handshake failed: x509: certificate signed by unknown authority\"","msg":"updater interrupted","severity":"info","ts":"2021-04-05T09:32:30.153631591Z"}
```

Also not sure if it is related but when I try to open osquery Kibana page, I am getting an error `Could not locate that index-pattern-field (id: result.name.keyword)`. If that is not related I will create a new thread for this.

**Reply by defensivedepth** [reply]
@facyber I see that you opened up a new Discussion with the same information (?) Please only post to one Discussion about your issue - it makes it too confusing otherwise, thanks!

**Reply by facyber** [reply]
I apologize,  as I also had issue with the Linux hosts, and that Kibana issue, I thought it's not the same issue. Would you like me to remove my previous reply or to just close Discussion that I opened?

**Reply by sleepingbel** [reply]
I don't remember in which topic this has appeared before on the discussion group, but I had the same problem. 
The FQDN was registered in the local DNS server. 

The only solotion was putting IP address instead of FQDN in the client config file.

**Reply by defensivedepth** [reply]
@facyber We can leave this reply, just post all details on that other Discussion, thanks!
